File: Lecture 4/StarterCode - Java/app/src/main/java/com/inovaeg/androidtrainingpharos2020/dataStorage/StorageExamplesActivity.java
package com.inovaeg.androidtrainingpharos2020.dataStorage;

import android.content.Intent;
import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

import com.inovaeg.androidtrainingpharos2020.R;

public class StorageExamplesActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_storage_examples);
        findViewById(R.id.bt_sharedPref).setOnClickListener(v -> {
            Intent intent = new Intent(this, SharedPrefActivity.class);
            intent.putExtra(BaseStorageActivity.TYPE_KEY, BaseStorageActivity.TYPE_SHARED_PREF);
            startActivity(intent);
        });
        findViewById(R.id.bt_dataBase).setOnClickListener(v -> {
            Intent intent = new Intent(this, DataBaseActivity.class);
            intent.putExtra(BaseStorageActivity.TYPE_KEY, BaseStorageActivity.TYPE_DATABASE);
            startActivity(intent);
        });
        findViewById(R.id.bt_file).setOnClickListener(v -> {
            Intent intent = new Intent(this, FileEampleActivity.class);
            intent.putExtra(BaseStorageActivity.TYPE_KEY, BaseStorageActivity.TYPE_FILE_SAVE);
            startActivity(intent);
        });
    }
}

File: Lecture 4/FinalCode - Kotlin/app/src/main/java/com/inovaeg/androidtrainingpharos2020/dataStorage/StorageExamplesActivity.kt
package com.inovaeg.androidtrainingpharos2020.dataStorage

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.inovaeg.androidtrainingpharos2020.R
import kotlinx.android.synthetic.main.activity_storage_examples.*

class StorageExamplesActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_storage_examples)
        bt_sharedPref.setOnClickListener { v: View? ->
            val intent = Intent(this, SharedPrefActivity::class.java)
            intent.putExtra(BaseStorageActivity.Companion.TYPE_KEY, BaseStorageActivity.Companion.TYPE_SHARED_PREF)
            startActivity(intent)
        }
        bt_dataBase.setOnClickListener { v: View? ->
            val intent = Intent(this, DataBaseActivity::class.java)
            intent.putExtra(BaseStorageActivity.Companion.TYPE_KEY, BaseStorageActivity.Companion.TYPE_DATABASE)
            startActivity(intent)
        }
        bt_file.setOnClickListener { v: View? ->
            val intent = Intent(this, FileEampleActivity::class.java)
            intent.putExtra(BaseStorageActivity.Companion.TYPE_KEY, BaseStorageActivity.Companion.TYPE_FILE_SAVE)
            startActivity(intent)
        }
    }
}